.circleci/deploy.py

```diff
@@ -0,0 +1,218 @@
+"""
+This script is run after initialize.py, and is reposible for deploying Review
+Apps on  your Dash Enterprise host
+"""
+
+import sys
+import subprocess
+from gql import gql, Client
+from gql.transport.requests import RequestsHTTPTransport
+from settings import (
+    SERVICE_PRIVATE_SSH_KEY,
+    DASH_ENTERPRISE_HOST,
+    APPNAME,
+    TARGET_APPNAME,
+    BRANCHNAME,
+    TRUNK_BRANCHNAME,
+    SERVICE_USERNAME,
+    SERVICE_API_KEY,
+)
+
+if sys.version_info[0] < 3.6 and sys.version_info[0] > 3.7:
+    raise Exception(
+        "This script has only been tested on Python 3.6."
+        + "You are using {version}.".format(version=sys.version_info[0])
+    )
+
+transport_service = RequestsHTTPTransport(
+    url=f"https://{DASH_ENTERPRISE_HOST}/Manager/graphql",
+    auth=(SERVICE_USERNAME, SERVICE_API_KEY),
+    use_json=True,
+    retries=5,
+)
+
+client_service = Client(transport=transport_service)
+
+
+def exit_message():
+    """
+    Prints out links to deployed app and app settings page before exiting
+    script.
+    """
+    print(
+        """
+        You Dash app has been deployed.
+
+        Preview {APPNAME}:
+
+        https://{DASH_ENTERPRISE_HOST}/{APPNAME}/
+        https://{DASH_ENTERPRISE_HOST}/Manager/apps/{APPNAME}/settings
+        """.format(
+            APPNAME=APPNAME,
+            DASH_ENTERPRISE_HOST=DASH_ENTERPRISE_HOST,
+        )
+    )
+    sys.exit()
+
+
+def zip_list_index(index_list, index_a, index_b):
+    """
+    Accepts two indices, from a single list, with values equivalent to a list
+    and zips them together, resulting in a dictionary.
+    """
+    index_key = [index_list[i][index_a] for i in range(len(index_list))]
+    index_value = [index_list[i][index_b] for i in range(len(index_list))]
+    return dict(zip(index_key, index_value))
+
+
+def handle_error(query_result, detected_error=None):
+    """
+    Raise error if error is not an accepted error
+    """
+    if detected_error is not None:
+        for accepted_error, error_message in accepted_errors.items():
```

Comment:
pass `accepted_errors` into the function to make this more explicit. Otherwise, it's difficult to read the code knowing that some variables are implicitly being used inside certain functions - readers usually expect that all variables will be passed into functions

Reply:
Same thing with `result`